aria_data_loaders/aria_data_utils/aria_live_streamer.py

```diff
@@ -0,0 +1,598 @@
+                img_frame=viz_img,
+                outputs=outputs,
+                device_T_marker=device_T_marker,
+                timestamp=frame.get("timestamp"),
+                img_metadata=None,
+            )
+
+        if detect_objects:
+            viz_img, object_scores = self.object_detector.process_frame(viz_img)
+            if object_label in object_scores.keys():
+                if self.verbose:
+                    plt.imsave(f"frame_{self._in_index}.jpg", viz_img)
+                self.publish_pose_of_interest(frame.get("ros_pose"))
+        self._in_index += 1
+        return viz_img, outputs
+
+    def initialize_april_tag_detector(self, outputs: dict = {}):
+        """
+        Initialize the april tag detector
+
+        Args:
+            outputs (dict, optional): Dictionary of outputs from the april tag detector. Defaults to {}.
+
+        Updates:
+            - self.april_tag_detector: AprilTagDetectorWrapper object
+
+        Returns:
+            outputs (dict): Dictionary of outputs from the april tag detector with following keys:
+                - "tag_image_list" - List of np.ndarrays of images with detections
+                - "tag_image_metadata_list" - List of image metadata
+                - "tag_device_T_marker_list" - List of Sophus SE3 transforms from Device frame to marker
+        """
+        self.april_tag_detector = AprilTagDetectorWrapper()
+        focal_lengths = self._dst_calib_params.get_focal_lengths()  # type:ignore
+        principal_point = self._dst_calib_params.get_principal_point()  # type: ignore
+        self.april_tag_detector.enable_detector()
+
+        outputs.update(
+            self.april_tag_detector._init_april_tag_detector(
+                focal_lengths=focal_lengths, principal_point=principal_point
+            )
+        )
+        return outputs
+
+    def initialize_object_detector(self, outputs: dict = {}, object_labels: list = []):
+        """
+        Initialize the object detector
+
+        Args:
+            outputs (dict, optional): Dictionary of outputs from the object detector. Defaults to {}.
+            object_labels (list, optional): List of object labels to detect. Defaults to [].
+
+        Updates:
+            - self.object_detector: ObjectDetectorWrapper object
+
+        Returns:
+            outputs (dict): Dictionary of outputs from the object detector with following keys:
+                - "object_image_list" - List of np.ndarrays of images with detections
+                - "object_image_metadata_list" - List of image metadata
+                - "object_image_segment" - List of Int signifying which segment the image
+                    belongs to; smaller number means latter the segment time-wise
+                - "object_score_list" - List of Float signifying the detection score
+        """
+        self.object_detector = ObjectDetectorWrapper()
+        self.object_detector.enable_detector()
+        meta_objects: List[str] = []
```

Comment:
what is meta_objects?

Reply:
In our OWL-ViT  based hand-object intersection logic, we ask OWL-ViT to detect a hand and the object of interest and then see if the bboxes intersect. "hand" then is the meta-object as it is something we provided, while core_object is the object_label passed by the user.

This was a noisy logic and we plan on replacing this with more sophisticated hand-object-interaction detectors.

Reply:
Let's add a comment about what meta-objects are for now?

Reply:
added